src/main/java/org/example/IntList.java

```diff
@@ -654,6 +666,34 @@ public Builder dropRight(int count) {
             return this;
         }
 
+        public Builder takeWhile(IntPredicate predicate) {
+            int i = start;
+
+            while (predicate.test(buffer[i]) && i < end) {
+                i++;
+                if (i == end) {
```

Comment:
This instruction and its condition (`i == end`) will the exit loop just before the loop would exit on itself (`i < end`). Moving this condition just after the loop would not change the behavior of the code while having any measurable impact on performance. Theses lines are probably not useful. @Glorfyndel